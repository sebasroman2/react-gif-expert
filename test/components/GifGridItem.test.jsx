import { render, screen } from "@testing-library/react"

import { GifGridItem } from "../../src/components/GifGridItem"

describe('Pruebas en GifGridItem', () => {
    const title = 'Saitana';
    const url = 'https://one-piece.com/luffy.jpg'

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifGridItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('Debe de mostrar el titulo en el componente', () => { 
        render(<GifGridItem title={title} url={url} />);
        expect( screen.getByText( title ) ).toBeTruthy();
     })
 })