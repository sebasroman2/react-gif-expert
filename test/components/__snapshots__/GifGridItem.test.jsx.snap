// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitana"
      src="https://one-piece.com/luffy.jpg"
    />
    <p>
      Saitana
    </p>
  </div>
</div>
`;
